apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt

```diff
@@ -0,0 +1,165 @@
+    }
+
+    private fun cleanUp() {
+        room?.disconnect()
+        room = null
+        localParticipantWrapper.participant = null
+        participants?.clear()
+    }
+
+    override fun onConnected(room: Room) {
+        Timber.i("onConnected -> room sid: %s", room.sid)
+
+        room.localParticipant?.let { localParticipant ->
+            participants = mutableListOf<ParticipantStream>().let { list ->
+                localParticipantWrapper.participant = localParticipant
+                list.add(ParticipantStream(localParticipantWrapper))
+
+                room.remoteParticipants.filter {
+                    !it.isVideoComposer()
+                }.forEach {
+                    val newRemoteParticipant = RemoteParticipantWrapper(it)
+                    list.add(ParticipantStream(newRemoteParticipant))
+                }
+                _onStateEvent.value = RoomViewEvent.OnConnected(list, room.name)
+                list
+            }
+        }
+        this.room = room
+    }
+
+    override fun onConnectFailure(room: Room, twilioException: TwilioException) {
+        this.localParticipantWrapper.participant = null
+        handleError(ErrorResponse(twilioException.message ?: "Error", twilioException.explanation ?: "onConnectFailure"))
+    }
+
+    override fun onReconnecting(room: Room, twilioException: TwilioException) {
+
+    }
+
+    override fun onReconnected(room: Room) {
+
+    }
+
+    override fun onDisconnected(room: Room, twilioException: TwilioException?) {
+
+    }
+
+    override fun onParticipantConnected(room: Room, remoteParticipant: RemoteParticipant) {
+        if (remoteParticipant.isVideoComposer()) return
+        val newParticipant = ParticipantStream(RemoteParticipantWrapper(remoteParticipant))
+        participants?.add(newParticipant)
+        _onStateEvent.value = RoomViewEvent.OnRemoteParticipantConnected(participants ?: listOf())
+    }
+
+    override fun onParticipantDisconnected(
+        room: Room,
+        remoteParticipant: RemoteParticipant
+    ) {
+        participants?.first { it.wrapper.participant?.identity == remoteParticipant.identity }?.apply {
+            participants?.remove(this)
+            _onStateEvent.value = RoomViewEvent.OnRemoteParticipantDisconnected(participants ?: listOf())
+        }
+    }
+
+    override fun onRecordingStarted(room: Room) {
+
+    }
+
+    override fun onRecordingStopped(room: Room) {
+
+    }
+
+    override fun onDominantSpeakerChanged(room: Room, remoteParticipant: RemoteParticipant?) {
+        super.onDominantSpeakerChanged(room, remoteParticipant)
+        participants?.first { it.wrapper.isDominantSpeaker }?.apply { this.wrapper.isDominantSpeaker = false }
+        participants?.first { it.wrapper.identity == remoteParticipant?.identity }?.apply {
+            this.wrapper.isDominantSpeaker = true
+        }
+    }
+
+    private fun RemoteParticipant.isVideoComposer(): Boolean =
```

Comment:
Interesting syntax!

Reply:
😄 yes, this is the way to extend functions for a specific already defined class.